server/http/restapi/call.go

```diff
@@ -10,17 +10,31 @@ import (
 	"github.com/mattermost/mattermost-plugin-apps/server/utils/httputils"
 )
 
-func (a *restapi) handleCall(w http.ResponseWriter, req *http.Request) {
+func (a *restapi) handleCallHTTP(w http.ResponseWriter, req *http.Request, sessionID, actingUserID string) {
 	call, err := apps.CallRequestFromJSONReader(req.Body)
 	if err != nil {
 		httputils.WriteError(w, utils.NewInvalidError(errors.Wrap(err, "failed to unmarshal Call request")))
 		return
 	}
 
-	res := a.proxy.Call(sessionID(req), actingID(req), call)
-	if res.Type == "" {
-		res.Type = apps.CallResponseTypeOK
+	res, err := a.handleCall(sessionID, actingUserID, call)
+	if err != nil {
+		httputils.WriteError(w, err)
+		return
 	}
 
 	httputils.WriteJSON(w, res)
 }
+
+func (a *restapi) handleCall(sessionID, actingUserID string, call *apps.CallRequest) (*apps.ProxyCallResponse, error) {
```

Comment:
Nit pick: Does a separate function make the code more readable?

Reply:
@hanzei @levb I wrote it this way so that it was easier to test. It made it so I didn't need to use requests or writers in the test. That being said, the test is actually less useful since it covers less code. Do you suggest I modify the test to use requests and writers instead of extracting this function?

Reply:
@hanzei I've put the functions back the way they were, and modified the tests to use http requests